### PR TITLE
Cleanup and update C# examples

### PR DESCRIPTION
**Pull request by jtattermusch**
Cleanup in the examples/csharp directory:

- make the helloworld example using dotnet cli the default (helloworld-from-cli -> Helloworld)
- mark the original helloworld using the legacy .csproj project as "still fully supported but legacy" (helloworld -> HelloworldLegacyCsproj)
- convert RouteGuide to the new-style .csproj projects (leads to simplification and easier maintenance)
- update the READMEs accordingly.

Note: After this is merged and the release is cut, I'll need to do a pass on grpc.io site to make sure the docs there capture the new directory names from this PR.

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,976,356      Total (=)      1,976,356

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,770,406      Total (=)     10,770,406

 No significant differences in binary sizes


```

File: examples/csharp/HelloworldLegacyCsproj/README.md [hunk 1]
```diff
@@ -3,6 +3,11 @@ gRPC in 3 minutes (C#)
 
 BACKGROUND
 -------------
+This is a different version of the helloworld example, using the old-style .csproj
+files supported by VS2013 and VS2015 (and older versions of mono).
+You can still use gRPC with the old-style .csproj files, but using the new-style
```
Comment (apolcyn) [inline examples/csharp/HelloworldLegacyCsproj/README.md]: nit: I might add to this sentence, e.g. "for instructions on using gRPC with new-style projects, see [../helloworld/READMD.md]" .... just to be clear that this example is for the old-style projects.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,976,610      Total (=)      1,976,610

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,771,520      Total (<)     10,771,522

 No significant differences in binary sizes


```

**Comment by jtattermusch** [comment]
Examples only change. Merging.

**Pull request merged**